Feature/enhance hipchat card

eloo opened the pull request:
This PR will fix #9110.
Adds thumbnail to HipChat notification
Add metric values to notification
Add message to notification

eloo committed:
Reorder editorconfig

eloo committed:
Add values to the hipchat card

eloo committed:
Add thumbnail to card

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/grafana/grafana?pullRequest=9111) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/grafana/grafana?pullRequest=9111) before we can accept your contribution.<br/><hr/>**Joseph Weigl** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/grafana/grafana?pullRequest=9111) it.</sub>

torkelo commented on pkg/services/alerting/notifiers/hipchat.go:
```diff
@@ -123,15 +123,24 @@ func (this *HipChatNotifier) Notify(evalContext *alerting.EvalContext) error {
 		"url":         ruleUrl,
 		"id":          "1",
 		"title":       evalContext.GetNotificationTitle(),
-		"description": evalContext.GetNotificationTitle() + " in state " + evalContext.GetStateModel().Text,
```
since you removed the GetStateModel().Text and it's not in the title, this will make the notification have info on state?

Eloo replied:
There is still the [Alerting] and [OK] prefix in the title and the custom message is now there too.
I think this is sufficient

torkelo replied:
yes, your right!

torkelo commented:
please resolve conflict and sign CLA :)

mihaiturcu commented:
any updates guys? :)

Eloo commented:
not yet. will have a look at it on the weekend

eloo committed:
Merge branch 'master' into feature/enhance_hipchat_card

Eloo commented:
conflicts resolved and CLA signed

mihaiturcu commented:
any timeline on when this might be released?

eloo commented:
why was the license check status changed?

bergquist commented:
I've been testing this PR and for some reason the notifier fails when going from alerting back to ok. Have you tested this?

eloo commented:
not sure,
will give this a try next week

eloo committed:
Fix empty message and toolong attribute names
Use default state message if no message is provided by the user
Slice attribute name to maximum of 50 chars

eloo commented:
@bergquist 
found the problem. the hipchat api will not allow empty messages no matter if the message will be displayed or not.
further i fixed an issue with long attribute/metric-names

bergquist commented on pkg/services/alerting/notifiers/hipchat.go:
```diff
@@ -123,15 +132,24 @@ func (this *HipChatNotifier) Notify(evalContext *alerting.EvalContext) error {
+		"date":       evalContext.EndTime.Unix(),
+		"attributes": attributes,
+	}
+	if len(evalContext.ImagePublicUrl) > 0 {
```
Please use `evalContext.ImagePublicUrl != ""` instead
Its the prefered way of checking for none empty values in Go

eloo replied:
done

bergquist commented on pkg/services/alerting/notifiers/hipchat.go:
```diff
@@ -144,6 +162,7 @@ func (this *HipChatNotifier) Notify(evalContext *alerting.EvalContext) error {
 
 	hipUrl := fmt.Sprintf("%s/v2/room/%s/notification?auth_token=%s", this.Url, this.RoomId, this.ApiKey)
 	data, _ := json.Marshal(&body)
+	this.log.Debug(string(data))
```
Its preferred to describe what your logging.
`this.log.Info("Request payload", "json", string(data))`

eloo replied:
done

bergquist commented on pkg/services/alerting/notifiers/hipchat.go:
```diff
@@ -84,32 +84,41 @@ func (this *HipChatNotifier) Notify(evalContext *alerting.EvalContext) error {
 		message += " " + evalContext.Rule.Message
 	}
+
+	if len(message) < 1 {
```
Please use `message == ""`
Its the prefered way of checking for empty values in Go

bergquist reviewed (changes requested):
This works fine! 

Just some minor changes and this is ready to be merged.

eloo committed:
Change empty string checks and improve logging

bergquist commented:
@eloo Great! Thank you for contributing! Will merge once CI tests passes :)
